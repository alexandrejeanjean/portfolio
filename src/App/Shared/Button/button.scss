@import '../../constants/colors.scss';

.classic-btn {
  font-size: 1.2rem;
  line-height: 35px;
  background-color: $fourth;
  color: #00b1da;
  background: linear-gradient(259.07deg, #daf8ff 4.07%, #b2eaf8 100%);
  box-shadow: 0px 8px 20px #e5f4f5;
  margin-top: 10px;
  margin-bottom: 40px;
  position: relative;
  padding: 0.25em 3em;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  border-radius: 5px;
  transition-duration: 0.5s;
  transition: all linear 0.3s;

  &::before {
    content: '';
    position: absolute;
    border-style: solid;
    border-width: 3px;
    border-color: $fourth;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    -webkit-animation-duration: 1s;
    animation-duration: 1s;
    border-radius: 10px;
    opacity: 0;
  }

  &:hover {
    text-decoration: none;
    transition-duration: 0.1s;
    box-shadow: 0px 8px 20px #daeff1;
  }
}
