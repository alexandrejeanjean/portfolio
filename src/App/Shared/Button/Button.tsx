import React from 'react'
import { Link } from 'react-router-dom'
import './button.scss'

type Props = { text: string; endpoint: string }

const ClassicButton = ({ text, endpoint }: Props) => (
  <Link className='classic-btn' to={endpoint}>
    {text}
  </Link>
)

export default ClassicButton
