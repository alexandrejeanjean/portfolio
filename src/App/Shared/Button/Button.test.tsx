import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it("should render button for 'a-propos' link", () => {
    const button = shallow(<Button text="Faisons connaissance" endpoint="/a-propos" />);
    expect(button).toMatchSnapshot();
  });
});