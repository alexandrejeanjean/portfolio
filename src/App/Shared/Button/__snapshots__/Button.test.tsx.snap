// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button for 'a-propos' link 1`] = `
<Link
  className="classic-btn"
  to="/a-propos"
>
  Faisons connaissance
</Link>
`;
